core/src/main/java/com/novoda/espresso/AccessibilityRules.java

```diff
@@ -0,0 +1,67 @@
+package com.novoda.espresso;
+
+import android.accessibilityservice.AccessibilityServiceInfo;
+import android.content.Context;
+import android.content.Intent;
+import android.support.test.InstrumentationRegistry;
+import android.view.accessibility.AccessibilityManager;
+
+import org.junit.rules.TestRule;
+
+import java.util.List;
+
+import static com.novoda.espresso.AccessibilityServiceToggler.Service;
+
+public class AccessibilityRules {
+
+    public static TestRule createTalkBackTestRule() {
+        return new AccessibilityServiceTestRule(createHelper(Service.TALKBACK));
+    }
+
+    public static TestRule createSwitchAccessTestRule() {
+        return new AccessibilityServiceTestRule(createHelper(Service.SWITCH_ACCESS));
+    }
+
+    public static TestRule createSelectToSpeakTestRule() {
+        return new AccessibilityServiceTestRule(createHelper(Service.SELECT_TO_SPEAK));
+    }
+
+    private static AccessibilityServiceTestRule.Helper createHelper(final Service service) {
+        final Context context = InstrumentationRegistry.getTargetContext();
+        final AccessibilityManager a11yManager = (AccessibilityManager) context.getSystemService(Context.ACCESSIBILITY_SERVICE);
+
+        return new AccessibilityServiceTestRule.Helper() {
+            @Override
+            public String serviceName() {
+                return service.name();
+            }
+
+            @Override
+            public void requestEnableService() {
+                requestState(AccessibilityServiceTogglingActivity.ENABLED);
+            }
+
+            @Override
+            public void requestDisableService() {
+                requestState(AccessibilityServiceTogglingActivity.DISABLED);
+            }
+
+            private void requestState(String value) {
+                Intent intent = new Intent(AccessibilityServiceTogglingActivity.ACTION_SET_SERVICE);
+                intent.putExtra(service.name(), value);
+                context.startActivity(intent);
+            }
+
+            @Override
+            public boolean reportsEnabled() {
+                List<AccessibilityServiceInfo> enabledAccessibilityServiceList = a11yManager.getEnabledAccessibilityServiceList(AccessibilityServiceInfo.FEEDBACK_SPOKEN);
+                for (AccessibilityServiceInfo accessibilityServiceInfo : enabledAccessibilityServiceList) {
+                    if (accessibilityServiceInfo.getId().equals(service.qualifiedName())) {
```

Comment:
1. it's not enough to check SecureSettings to see if the value is set - probably there is an observer on that file which then enables the service, so we must query the a11y manager.
2. we're querying by service id. this is not cool in production apps because we don't want to tie our implementations to specific accessibility services; instead, we want to tie implementations to abstractions like a spoken feedback service. it's not sufficient in this case because we want to check whether the exact service is enabled/disabled.